src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -244,6 +244,22 @@ def set_halt(self) -> None:
         self.set_shutdown()
         self.stop()
 
+    @classmethod
+    def try_add_heartbeat(cls, raw: Dict) -> bool:
+        try:
+            entry = NodeHeartbeatEntry.parse_obj(raw)
```

Comment:
The parsing should be done as close to ingest as possible, which should be at the function level.